cardano-api/src/Cardano/Api/Script.hs

```diff
@@ -390,6 +390,11 @@ data Script lang where
 deriving instance (Eq   (Script lang))
 deriving instance (Show (Script lang))
 
+instance Ord (Script lang) where
+  -- versions must be the same
+  compare (SimpleScript _ sA) (SimpleScript _ sB) = compare sA sB
+  compare (PlutusScript _ sA) (PlutusScript _ sB) = compare sA sB
```

Comment:
Double check that you _really_ need all these new tricky `Ord` instances. I'm moderately confident that you could do it all with just `AnyScriptLanguage` which already has an `Ord` instance (where it naturally makes sense since that type covers all the languages and their versions).

Reply:
we need to unify
```hs
, txAuxScripts =
  TxAuxScripts AuxScriptsInAlonzoEra
    [ ScriptInEra
        PlutusScriptV1InAlonzo
        (PlutusScript PlutusScriptV1 (PlutusScriptSerialised ""))
    , ScriptInEra
        SimpleScriptV2InAlonzo
        (SimpleScript SimpleScriptV2 (RequireSignature "ebe9de78a37f84cc819c0669791aa0474d4f0a764e54b9f90cfe2137"))
    ]
```
and
```hs
, txAuxScripts =
  TxAuxScripts AuxScriptsInAlonzoEra
    [ ScriptInEra
        SimpleScriptV2InAlonzo
        (SimpleScript SimpleScriptV2 (RequireSignature "ebe9de78a37f84cc819c0669791aa0474d4f0a764e54b9f90cfe2137"))
    , ScriptInEra
        PlutusScriptV1InAlonzo
        (PlutusScript PlutusScriptV1 (PlutusScriptSerialised ""))
    ]
```
What is a proper way to do it?

Reply:
nevermind, done